docs/development/design/apiv3.rst

```diff
@@ -20,185 +39,117 @@ Problems with APIv2
+This iteration will include:
 
+* Minor changes to fields returned in the objects
+* Import Project endpoint
+* Edit Project attributes ("Settings" and "Advanced settings-Global settings" in the WebUI)
+* Trigger Build for default version
+* Allow CRUD for Redirect, Environment Variables and Notifications (``WebHook`` and ``EmailHook``)
+* Documentation
 
-Proper status codes for error reporting
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-* Pros:
-
-  * user knows what it's happening and can take decisions about it
-
-* Cons:
-
-  * n/a
-
-
-Relationship between API resources
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-* Pros:
-
-  * browse-able API by accessing to the default resource (Project)
-  * knowing the project's slug you can perform all the actions related to it
-
-* Cons:
-
-  * more data is returned with all the links to the endpoints
-
-
-Make footer API returns JSON
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-* Pros:
-
-  * users can build their own version menu flyout
-  * cleaner than returning a HTML and injecting it
-
-* Cons:
-
-  * we need to adapt our theme for this
-
-
-Questions:
-
-#. Do we want an specific endpoint for the footer?
-#. The flyout could be built by querying 2 or 3 of the regular
-   endpoint. Would this add too much overhead to the page loading
-   time?
-
-
-Use cases
----------
 
-We want to cover
-++++++++++++++++
+Version 3
++++++++++
 
+Third iteration will implement granular permissions.
+Keeping in mind how Sphinx extension will use it:
 
-* Return all **my** projects
-* Access all the resources by knowing the project's slug
-* Ability to filter by fields
+* ``sphinx-version-warning`` needs to get *all active Versions of a Project*
+* An extension that creates a landing page, will need *all the subprojects of a Project*
 
-  * all the active versions of specific project
+To fulfill these requirements, this iteration will include:
 
-* Data about builds
+* Scope-based authorization token
 
-  * latest build for project
-  * latest build for a particular version of a project
-  * status of a particular build
 
-* Perform write actions like
+Version 4
++++++++++
 
-  * add a Domain,
-  * add User as mantainer,
-  * import a new Project under my username,
-  * set the language of the Project,
-  * trigger a Build,
-  * activate/deactivate a Version to be built,
-  * and all the actions you can perform from the Admin tab.
+* Specific endpoint for our flyout menu (returning JSON instead of HTML)
 
-* Retrieve all the information needed to create a custom version menu flyout
 
+Out of roadmap
+--------------
 
-Considering some useful cases for the corporate site:
+These are some features that we may want to build but they are not in the roadmap at this moment:
 
-* Give access to a doc page (``objects.inv``, ``/design/core.html``)
+* CRUD for Domain
+* Add User as maintainer
+* Give access to a documentation page (``objects.inv``, ``/design/core.html``)
+* Internal Build process
```

Comment:
We should mention why all of these are out of the roadmap.

Reply:
I put them out of the roadmap for now because, in case we build them, we need to decide in which Version they will be assigned. Also, they do not affect too many users or are only for internal usage only.

I'm updating the paragraph mentioning this.